tokio-postgres/src/passfile/mod.rs

```diff
@@ -0,0 +1,199 @@
+//! Support for reading password files.
+//!
```

Comment:
Can you add some more documentation here about compatibility with libpq and various choices you made about corner cases and type representations?

For example, postgres supports many encodings, so we don't know whether the representation of a username is UTF8 or something else, so we have to use bytes.

Are there differences in how libpq vs this PR handle malformed files?

Reply:
I've added some more text.

I've tried to imitate what libpq does for most malformed files. Unsurprisingly libpq behaves strangely if the file has embedded NULs. I haven't tried to reproduce that but I've included a note about it.

libpq also has a bug where if the file lacks a trailing newline the final line, which would normally be processed, is ignored if it exactly fills the read buffer. I don't think that's worth mentioning.